sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateEncoder.scala

```diff
@@ -292,7 +299,13 @@ class RangeKeyScanStateEncoder(
       // Note that we cannot allocate a smaller buffer here even if the value is null
       // because the effective byte array is considered variable size and needs to have
       // the same size across all rows for the ordering to work as expected.
-      val bbuf = ByteBuffer.allocate(field.dataType.defaultSize + 1)
+      val bbuf = if (!supportsSignedValues(field.dataType)) {
+        ByteBuffer.allocate(field.dataType.defaultSize + 1)
+      } else {
+        // for numeric types, we reserve 2 additional bytes. one to indicate if the value is null
+        // and the other to indicate if the value is negative.
+        ByteBuffer.allocate(field.dataType.defaultSize + 2)
+      }
```

Comment:
Can we just use one byte to indicate sign and Null. If we use `0x00` for -ve, `0x01` for +ve and `0x02` for Null. we should be able to save a byte per column.

Reply:
Done